crates/primitives/starknet/src/crypto/hash/pedersen.rs

```diff
@@ -20,10 +21,27 @@ impl HasherT for PedersenHasher {
     /// # Returns
     /// The hash of the data.
     fn hash_bytes(&self, data: &[u8]) -> Felt252Wrapper {
-        // For now we use the first 31 bytes of the data as the field element, to avoid any panics.
-        // TODO: have proper error handling and think about how to hash efficiently big chunks of data.
-        let field_element = FieldElement::from_byte_slice_be(&data[..31]).unwrap();
-        Felt252Wrapper(pedersen_hash(&FieldElement::ZERO, &field_element))
+        // Calculate the number of 31-byte chunks we'll need, rounding up.
+        // (1 byte is used padding to prevent the value of field from being greater than modular)
+        // TODO: It is need a way to truncate bytes to fit into values smaller than modular (optimization)
+        let chunks = (data.len() + 30) / 31;
+
+        let mut hash_value = FieldElement::ZERO;
+
+        for i in 0..chunks {
```

Comment:
could we use the `.chunks` method here ?

Reply:
Thanks for the advice, I've found that implementing it using the `.chunks` that exist in std::slice will result in more concise code. Do you mind if I ask you to review it one more time?